src/main/java/seedu/academyassist/logic/parser/AddClassCommandParser.java

```diff
@@ -33,16 +32,13 @@ public AddClassCommand parse(String args) throws ParseException {
 
         argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_IC, PREFIX_SUBJECT); // Only can add one class at a time.
         Ic ic = ParserUtil.parseIc(argMultimap.getValue(PREFIX_IC).get());
-        Set<Subject> subjects = ParserUtil.parseSubjects(argMultimap.getAllValues(PREFIX_SUBJECT));
 
-        // Verify that there is exactly one subject
-        if (subjects.size() != 1) {
-            throw new ParseException("Exactly one subject must be provided.");
+        try {
+            Subject subject = ParserUtil.parseSubject(argMultimap.getValue(PREFIX_SUBJECT).get());
+            return new AddClassCommand(ic, subject);
```

Comment:
Maybe we could consider changing `AddClassCommand` to `AddSubjectCommand` to reduce confusion and inconsistency?